src/query/plan/rule_based_planner.hpp

```diff
@@ -180,11 +180,16 @@ class RuleBasedPlanner {
         }
       }
       uint64_t merge_id = 0;
-      for (auto *clause : query_part.remaining_clauses) {
+      bool handle_empty_result = true;
```

Comment:
Can you describe the logic you implemented here? I don't see why for example a `MATCH` as a last clause should prevent the empty result. Also, shouldn't this be only relevant for the last clause of the last query part?

Reply:
Yes so it isn't relevant only to the last clause of the last query part because of the with, unwind and load csv clause. In this way we avoid having `EmptyResult` operator in the middle of the query plan.